compare_pkglist.py

```diff
@@ -0,0 +1,129 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+import logging
+import sys
+import urllib2
+import re
+from xml.etree import cElementTree as ET
+
+import osc.conf
+import osc.core
+
+from osc import oscerr
+
+OPENSUSE = 'openSUSE:Leap:42.2'
+SLE = 'SUSE:SLE-12-SP2:GA'
+
+makeurl = osc.core.makeurl
+http_GET = osc.core.http_GET
+http_POST = osc.core.http_POST
+http_PUT = osc.core.http_PUT
+
+class CompareList(object):
+    def __init__(self, old_prj, new_prj, verbose):
+        self.new_prj = new_prj
+        self.old_prj = old_prj
+        self.verbose = verbose
+        self.factory = 'openSUSE:Factory'
+        self.apiurl = osc.conf.config['apiurl']
+        self.debug = osc.conf.config['debug']
+
+    def get_source_packages(self, project, expand=False):
+        """Return the list of packages in a project."""
+        query = {'expand': 1} if expand else {}
+        root = ET.parse(http_GET(makeurl(self.apiurl,['source', project],
+                                 query=query))).getroot()
+        packages = [i.get('name') for i in root.findall('entry')]
+        
+        return packages
+
+    def check_diff(self, package, old_prj, new_prj):
+        logging.debug('checking %s ...' % package)
+        query = {'cmd': 'diff',
+                 'view': 'xml',
+                 'oproject': old_prj,
+                 'opackage': package}
+        u = makeurl(self.apiurl,['source', new_prj, package], query=query)
+        root = ET.parse(http_POST(u)).getroot()
+        old_srcmd5 = root.findall('old')[0].get('srcmd5')
+        logging.debug('%s old srcmd5 %s in %s' % (package, old_srcmd5, old_prj))
+        new_srcmd5 = root.findall('new')[0].get('srcmd5')
+        logging.debug('%s new srcmd5 %s in %s' % (package, new_srcmd5, new_prj))
+        # Compare srcmd5
+        if old_srcmd5 != new_srcmd5:
+            # check if it has diff element
+            diffs = root.findall('files/file/diff')
+            if diffs:
+                return ET.tostring(root)
+        return False
+
+    def crawl(self):
+        """Main method"""
+        # get souce packages from target
+        print 'Gathering the packages list from %s' % self.old_prj
+        old_packages = self.get_source_packages(self.old_prj)
+        print 'Gathering the packages list from %s' % self.new_prj
+        new_packages = self.get_source_packages(self.new_prj)
+
+        for pkg in old_packages:
+            have_diff = False
+            if pkg not in new_packages:
+                logging.debug('%s is not in %s' % (pkg, self.new_prj))
+            else:
+                have_diff = self.check_diff(pkg, self.old_prj, self.new_prj)
+                if have_diff is not False:
+                    print '%s/%s has different source than %s' % (self.new_prj, pkg, self.old_prj)
+                    if self.verbose:
+                        print have_diff
```

Comment:
It was not clear when reading this that `have_diff` actually contained the diff or `False`. It seemed it would print out `True/False` until digging deeper. Perhaps just `diff` as variable name?